crates/wit-bindgen/src/lib.rs

```diff
@@ -1757,41 +1744,18 @@ impl<'a> InterfaceGenerator<'a> {
 
         self.rustdoc(&func.docs);
 
-        let mut resources = Vec::new();
-        let mut args = String::new();
-
-        for (i, param) in func.params.iter().enumerate() {
-            uwrite!(args, "arg{}: ", i);
-            resources.append(&mut self.print_ty_(
-                &mut args,
-                &param.1,
-                TypeMode::AllBorrowed("'_"),
-                resources.len(),
-                true,
-            ));
-            args.push_str(",");
-        }
-
-        let resource_generics = resources
-            .iter()
-            .map(|(arg_name, _resource_name)| format!("{}: 'static", arg_name))
-            .collect::<Vec<String>>()
-            .join(", ");
-
         uwrite!(
             self.src,
-            "pub {async_} fn call_{}<",
+            "pub {async_} fn call_{}<S: wasmtime::AsContextMut>(&self, mut store: S, ",
             func.name.to_snake_case(),
         );
 
-        if !resources.is_empty() {
-            uwrite!(self.src, "{resource_generics}, ");
+        for (i, param) in func.params.iter().enumerate() {
+            uwrite!(self.src, "arg{}: ", i);
+            self.print_ty(&param.1, TypeMode::AllBorrowed("'_"));
```

Comment:
This will result in a `wasmtime::component::ResourceAny` when it should be `wasmtime::component::Resource<T>`? 
It seems to be the case in in exported functions taking an imported resource as an argument at least.

Given:
```wit
package example2:component@0.1.0

interface backend {
    resource scalars {
        constructor()
        get-b: func() -> u32

    }
}

interface front {
    use backend.{scalars}

    handle: func(x: own<scalars>) -> u32
}

world example2 {
    import backend
    export front
}
```

Will produce the following `call_handle` function:
```rust
                    pub fn call_handle<S: wasmtime::AsContextMut>(
                        &self,
                        mut store: S,
                        arg0: wasmtime::component::ResourceAny,
                    ) -> wasmtime::Result<u32> {
                        let callee = unsafe {
                            wasmtime::component::TypedFunc::<
                                (wasmtime::component::ResourceAny,),
                                (u32,),
                            >::new_unchecked(self.handle)
                        };
                        let (ret0,) = callee.call(store.as_context_mut(), (arg0,))?;
                        callee.post_return(store.as_context_mut())?;
                        Ok(ret0)
                    }
```
where I would expect the function signature to look like this?
```rust
pub fn call_handle<S: wasmtime::AsContextMut>(
                        &self,
                        mut store: S,
                        arg0: wasmtime::component::Resource<Scalars>,
                    ) -> wasmtime::Result<u32>;
```